src/controller/CHIPDeviceController.cpp

```diff
@@ -2316,23 +2316,23 @@ CHIP_ERROR DeviceController::UpdateDevice(NodeId peerNodeId)
 {
     VerifyOrReturnError(mState == State::Initialized && mFabricIndex != kUndefinedFabricIndex, CHIP_ERROR_INCORRECT_STATE);
 
-    OperationalDeviceProxy * proxy = GetDeviceSession(GetPeerScopedId(peerNodeId));
+    OperationalSessionSetup * proxy = GetDeviceSession(GetPeerScopedId(peerNodeId));
     VerifyOrReturnError(proxy != nullptr, CHIP_ERROR_NOT_FOUND);
 
     return proxy->LookupPeerAddress();
```

Comment:
Why is this a useful operation?  It's not going to affect anything, unlike the code that used to be here, right?

This looks to me like it breaks the "re-resolve the address" error recovery mechanism we have right now.  How does that mechanism work in the new setup?

Reply:
`UpdateDevice` is going to be removed in https://github.com/project-chip/connectedhomeip/issues/20452.

TBH I am not 100% sure how we re-resolve the address, but I was under the impression that the actual session gets marked as defunct and a new session will do a new address look up.

Reply:
We re-resolve the address after our first "we didn't get an ack".  We don't mark the session defunct until all MRP resends failed.

Reply:
That MRP is not going through this `UpdateDevice` path though, `UpdateDevice` is depreciated and should be removed in my follow up PR (that I hope will be ready for review within 1 to 2 days of me getting this PR in). But I do agree that this PR does break `OnFirstMessageDeliveryFailed`, that is a separate discussion

Reply:
> That MRP is not going through this UpdateDevice path though

It is before this PR!

Reply:
I got sidetracked by CI issue for this PR on crique so I didn't message the slack chat group yet.

Yeah we are effectively removing it. Would you be okay with me addressing the issue of `OnFirstMessageDeliveryFailed` in a follow up PR?

Based on the spec it says that we `SHOULD` do an address look up, and at this point this is a convince function that we can put back in to keep churn down in this PR lower

Reply:
I am definitely OK with this being addressed in a followup, as long as we have a specific plan for doing so, described in an issue tracking that happening.

Reply:
Here is the github issue and possible solutions. https://github.com/project-chip/connectedhomeip/issues/21639

We can have follow up discussions over there as to the best solution. Let me know if that is good enough for you